cmd/validate/image.go

```diff
@@ -492,10 +492,41 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 				}
 
 				// Process all VSAs using the service
-				err = vsaService.ProcessAllVSAs(cmd.Context(), report, getGitURL, getDigest)
+				vsaResult, err := vsaService.ProcessAllVSAs(cmd.Context(), report, getGitURL, getDigest)
 				if err != nil {
 					log.Errorf("Failed to process VSAs: %v", err)
 					// Don't return error here, continue with the rest of the command
+				} else {
+					// Upload VSAs to configured storage backends
+					if len(data.vsaUpload) > 0 {
+						log.Infof("[VSA] Starting upload to %d storage backend(s)", len(data.vsaUpload))
+
+						// Upload component VSA envelopes
+						for imageRef, envelopePath := range vsaResult.ComponentEnvelopes {
+							uploadErr := vsa.UploadVSAEnvelope(cmd.Context(), envelopePath, imageRef, data.vsaUpload, signer)
```

Comment:
Is `imageRef` necessary? It seems all it's used for is to log what type was uploaded, which can be done here if there's no error.

Reply:
> since the envelopes are already written to disk when they're created

Oh yes, that's a good point. I've removed the `imageRef` param from the `Upload` functions now since it was doing redundant work. Thanks!